src/core/pusher.ts

```diff
@@ -116,26 +132,46 @@ export default class Pusher {
           channel.handleEvent(event);
         }
       }
+
       // Emit globally [deprecated]
       if (!internal) {
         this.global_emitter.emit(event.event, event.data);
       }
     });
+
     this.connection.bind('connecting', () => {
       this.channels.disconnect();
     });
+
     this.connection.bind('disconnected', () => {
       this.channels.disconnect();
     });
+
     this.connection.bind('error', err => {
       Logger.warn(err);
     });
+  }
 
-    Pusher.instances.push(this);
-    this.timeline.info({ instances: Pusher.instances.length });
+  /**
+   * Allows you to switch Pusher cluster without
+   * losing all the channels/subscription binding
+   * as this is internally managed by the SDK.
+   */
+  switchCluster(options: ClusterOptions) {
+    // Automatically disconnect the current connection
+    this.disconnect();
 
-    this.user = new UserFacade(this);
+    const { appKey, cluster } = options;
```

Comment:
Should I manually validate the shape of the options as you're doing on the constructor?